### ISSUE TITLE
Bug: aws_lambda_powertools.utilities.parser.pydantic does not exist anymore while mentioned in the documentation

### ISSUE THREAD

**Issue by VladimirSvoboda** [awslabs/aws-lambda-powertools-python#5390]
### Expected Behaviour

Contrarily to what is stated in the [documentation](https://docs.powertools.aws.dev/lambda/python/latest/utilities/parser/#faq), since the version 3 (and more precisely the pull request #4502) , it is not possible anymore to import any Pydantic module/function through `aws_lambda_powertools.utilities.parser.pydantic`.

In my case, I would like to be able to import the decorator `@field_serializer`

### Current Behaviour

```
python -c "from aws_lambda_powertools.utilities.parser.pydantic import field_serializer"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'aws_lambda_powertools.utilities.parser.pydantic'
```

### Code snippet

python3 -m venv venv
source venv/bin/activate
pip install aws-lambda-powertools[all]==3.1.0
python -c "from aws_lambda_powertools.utilities.parser.pydantic import field_serializer"

### Possible Solution

1. Restore the file `aws_lambda_powertools/utilities/parser/pydantic.py` removed in #4502 
2. Correct the documentation

### Steps to Reproduce

Execute the following code snippet

```
python3 -m venv venv
source venv/bin/activate
pip install aws-lambda-powertools[all]==3.1.0
python -c "from aws_lambda_powertools.utilities.parser.pydantic import field_serializer"
```

### Powertools for AWS Lambda (Python) version

3.1.0

### AWS Lambda function runtime

3.12

### Packaging format used

PyPi

**Comment by leandrodamascena** [comment]
Hey @VladimirSvoboda! thanks a lot for opening this issue! In fact we are not exporting Pydantic objects from parser and we explain that in our [upgrade guide](https://docs.powertools.aws.dev/lambda/python/latest/upgrade/#importing-pydantic-objects) to v3.

In Powertools v3 we have dropped support for Pydantic v1 and are only supporting Pydantic v2 which is why we are no longer exporting pydantic types directly and allowing customers to import them directly from Pydantic. First, I must say that I agree with that we need to update our documentation and we are https://github.com/aws-powertools/powertools-lambda-python/pull/5262. I'll try to merge this PR as soon as possible to prevent this type of error.

Let me explain some reasons why we removed this export:

1 - Pydantic V2 was a complete refactoring of the Pydantic project and they made several changes including, but not limited, to the way they export types, classes, methods, and others. If we continue to expose them directly from Powertools and some export in Pydantic has changed, our customers may get the impression that there is some bug in Powertools and that is not the experience we want to have for our customers.

2 - The way we were doing this was by [exposing everything](https://github.com/aws-powertools/powertools-lambda-python/blob/v2/aws_lambda_powertools/utilities/parser/pydantic.py) exposed from pydantic to our customers, but there is a long discussion in the Pydantic repository about how https://github.com/pydantic/pydantic/discussions/6748 when importing and loading things in this new version. We don't want our customers to inadvertently import everything from Pydantic and this could create some slowdown in their functions.

We are working to improve this documentation in this PR - https://github.com/aws-powertools/powertools-lambda-python/pull/5262

Can you please try importing them directly from Pydantic?

Thanks

**Comment by VladimirSvoboda** [comment]
Thanks for your answer. I now import directly from pydantic to fix the issue.
Please note that the error regarding these imports in the documentation is currently *not* fixed by #5262

**Comment by anafalcao** [comment]
Hi @VladimirSvoboda ! Thank you for your follow-up and for bringing this to our attention. You're absolutely right that the documentation issue hasn't been fully addressed by [#526](https://github.com/aws-powertools/powertools-lambda-python/pull/5262) yet. We appreciate your patience and the workaround you've implemented using direct imports from pydantic.

We're actively working on a comprehensive update to the Parser documentation, which will include fixing the import errors you've mentioned. We expect to release these changes within the next few weeks. In the meantime, if you encounter any other issues, please don't hesitate to let us know. 
Thanks!

**Comment by leandrodamascena** [comment]
Hey @VladimirSvoboda! We are closing this issue because we merged the PR! We are releasing next week with this fix.
